pkg/services/app/add.go

```diff
@@ -120,6 +120,13 @@ func (a *App) Add(params AddParams) error {
 		return fmt.Errorf("could not update parameters: %w", err)
 	}
 
+	if params.SourceType != string(wego.SourceTypeHelm) {
+		err = a.git.ValidateAccess(ctx, params.Url, params.Branch)
+		if err != nil {
+			return err
```

Comment:
this is already wrapped up inside the `ValidateAccess` method

Reply:
True, but if validate access is called from multiple places then it could be hard to track down where exactly the error occurred.  I'm not sure what the standard practice is here, but wrapping it here with context that it is being called from add like `return fmt.Errorf("app add ValidateAccess %w");`

Reply:
Also, should the if-statement be `params.SourceType == wego.SourceTypeGit`?  If we add new source types besides helm and git in the future then this will ValidateAccess on those new source types.  This seems very specific to git.